### ISSUE TITLE
admin dashboard (self-hosted)

### ISSUE THREAD

**Issue by loban-v-navek** [hoppscotch/hoppscotch#3270]
hi. help me please. i signed in by magic link and got it. how become admin? it's self-hosted installation. i have no problem with users page

![1](https://github.com/hoppscotch/hoppscotch/assets/118425382/aa096ea9-7bc9-47b5-bc6b-de78c10d1318)

**Comment by liyasthomas** [comment]
Only an existing Admin can elevate a normal user to an Admin role. Ask your administrator to make yourself an Admin.

If you're already an Admin, make sure you're using the correct email ID that has Admin access to login into the Dashboard.

**Comment by loban-v-navek** [comment]
> Only an existing Admin can elevate a normal user to an Admin role. Ask your administrator to make yourself an Admin.
> 
> If you're already an Admin, make sure you're using the correct email ID that has Admin access to login into the Dashboard.

where i must specify admin? i build this installation but i don't have admin rights)

**Comment by liyasthomas** [comment]
The first email you used to sign in to the dashboard will have Admin privileges. Can you confirm you're using the same email to sign in to the dashboard?

**Comment by loban-v-navek** [comment]
> The first email you used to sign in to the dashboard will have Admin privileges. Can you confirm you're using the same email to sign in to the dashboard?

i can't. where i can find what email it was?

**Comment by balub** [comment]
@loban-v-navek, did you sign in to the hoppscotch app before you tried to log in to the admin app?

**Comment by loban-v-navek** [comment]
> @loban-v-navek, did you sign in to the hoppscotch app before you tried to log in to the admin app?

sorry but I can't answer. it was problem with magic link and may be it was hopp app before admin app

**Comment by loban-v-navek** [comment]
i changed some details
```
#-----------------------Backend Config------------------------------#
# Prisma Config
DATABASE_URL=postgresql://postgres:pass@hoppscotch-db:5432/hoppscotch

# Auth Tokens Config
JWT_SECRET="secret"
TOKEN_SALT_COMPLEXITY=10
MAGIC_LINK_TOKEN_VALIDITY= 3
REFRESH_TOKEN_VALIDITY="604800000" # Default validity is 7 days (604800000 ms) in ms
ACCESS_TOKEN_VALIDITY="86400000" # Default validity is 1 day (86400000 ms) in ms
SESSION_SECRET='secret'

# Hoppscotch App Domain Config
REDIRECT_URL="https://api.mydomain.com"
WHITELISTED_ORIGINS = "https://api.mydomain.com/graphql,https://api.mydomain.com/v1,https://api.mydomain.com,https://apiadm.mydomain.com"

# Google Auth Config
GOOGLE_CLIENT_ID="************************************************"
GOOGLE_CLIENT_SECRET="************************************************"
GOOGLE_CALLBACK_URL="http://localhost:3170/v1/auth/google/callback"
GOOGLE_SCOPE="email,profile"

# Github Auth Config
GITHUB_CLIENT_ID="************************************************"
GITHUB_CLIENT_SECRET="************************************************"
GITHUB_CALLBACK_URL="http://localhost:3170/v1/auth/github/callback"
GITHUB_SCOPE="user:email"

# Microsoft Auth Config
MICROSOFT_CLIENT_ID="************************************************"
MICROSOFT_CLIENT_SECRET="************************************************"
MICROSOFT_CALLBACK_URL="http://localhost:3170/v1/auth/microsoft/callback"
MICROSOFT_SCOPE="user.read"

# Mailer config
MAILER_SMTP_URL="smtp://hoppscotch@mydomain.com:pass@smtp.mydomain.com"
MAILER_ADDRESS_FROM='"hoppscotch mail" <hoppscotch@mydomain.com>'

# Rate Limit Config
RATE_LIMIT_TTL=60 # In seconds
RATE_LIMIT_MAX=100 # Max requests per IP


#-----------------------Frontend Config------------------------------#


# Base URLs
VITE_BASE_URL=https://api.mydomain.com
VITE_SHORTCODE_BASE_URL=https://api.mydomain.com
VITE_ADMIN_URL=https://apiadm.mydomain.com

# Backend URLs
VITE_BACKEND_GQL_URL=https://api.mydomain.com/graphql
VITE_BACKEND_WS_URL=wss://api.mydomain.com/graphql
VITE_BACKEND_API_URL=https://api.mydomain.com/v1

# Terms Of Service And Privacy Policy Links (Optional)
VITE_APP_TOS_LINK=https://docs.hoppscotch.io/support/terms
VITE_APP_PRIVACY_POLICY_LINK=https://docs.hoppscotch.io/support/privacy
```

**Comment by philippos86** [comment]
Thanks for the question and the answer.
I was facing the same issue acutally. 
I hosted hopscotch locally and run it directly (Not from the Docker container).
Eventually as @AndrewBastin mentioned, I accessed my DB using pgAdmin and changed `isAdmin` to true.
Still wondering though if this could have been done during the first installation (assigning the admin role to the first user by defauly), or should the installer do this manually every time.
